R/ard_continuous.R

```diff
@@ -233,28 +195,28 @@ ard_continuous <- function(data,
 
 #' Process Statistic Labels
 #'
-#' @param stat_labels named list
-#'
+#' @param stat_arg_list named list
+#' @param col_name column name in the ARD to make for the stat argument
 #' @return named list
 #' @keywords internal
 #' @examples
 #' list(AGE = list(c("N", "n") ~ "{n} / {N}")) |>
-#'   cards:::.process_stat_labels()
-.process_stat_labels <- function(stat_labels){
+#'   cards:::.process_stat_arg(col_name = "stat_label")
+.process_stat_arg <- function(stat_arg_list, col_name){
```

Comment:
This function uses `compute_formula_selector(data = NULL, x = x)`. Do you think it would be good to add another argument to pass a named list that we could use tidyselect selectors?

``` r
library(cards)

statistics <- compute_formula_selector(ADSL["AGE"], x = ~continuous_variable_summary_fns(c("median", "mean")))
statistics
#> $AGE
#> $AGE$median
#> function(x) stats::median(x, na.rm = TRUE)
#> <bytecode: 0x1155ff0f0>
#> <environment: 0x11560bf60>
#> 
#> $AGE$mean
#> function(x) mean(x, na.rm = TRUE)
#> <bytecode: 0x1155fd3c8>
#> <environment: 0x11560bf60>

# this is the part that would go in the function, so we could use tidyselect selectors
compute_formula_selector(
  data = statistics[["AGE"]], 
  x = everything() ~ "xx.xx"
)
#> $median
#> [1] "xx.xx"
#> 
#> $mean
#> [1] "xx.xx"
```

<sup>Created on 2023-12-18 with [reprex v2.0.2](https://reprex.tidyverse.org)</sup>

Reply:
This is a great idea, can do! (actually it is why i had the leftover `y` argument from a `map2` above...I was playing with that concept..)

Reply:
Oh nice! I am going to respond to two comments in this thread:

> This is a great idea, can do! (actually it is why i had the leftover y argument from a map2 above...I was playing with that concept..)

> Would you expect that it combines user-supplied with the `default_fmt_fn()` in this case? Maybe an update to `fill_formula_selectors`?

Let's implement the feature above so we can use tidyselect, then update the default value of `fmt_fn` to specify all the stats, something like 

```r
fmt_fn = everything() ~ default_fmt_fn()
```

where (this may need some tweaking 

```r
default_fmt_fns <- function(){
  list(
    everything() ~ function(x) format(round5(x, digits = 1), nsmall = 1), # this is the new entry
    n = function(x) format(round5(x, digits = 0), nsmall = 0),
    N = function(x) format(round5(x, digits = 0), nsmall = 0),
    N_obs = function(x) format(round5(x, digits = 0), nsmall = 0),
    N_miss = function(x) format(round5(x, digits = 0), nsmall = 0),
    N_nonmiss = function(x) format(round5(x, digits = 0), nsmall = 0),
    p = function(x) format(round5(x * 100, digits = 1), nsmall = 1),
    p_cell = function(x) format(round5(x * 100, digits = 1), nsmall = 1),
    p_miss = function(x) format(round5(x * 100, digits = 1), nsmall = 1),
    p_nonmiss = function(x) format(round5(x * 100, digits = 1), nsmall = 1)
  )
}
```

Reply:
With a default value that assigns every statistic, the `fill_formula_selectors()` will work better to fill any non-specified summary stats when a user begins modifying the defaults.

Reply:
I like the idea! I think I am still confused whether this is possible with the current logic in `fill_formula_selectors()`, specifically this line: https://github.com/insightsengineering/cards/blob/main/R/process_selectors.R#L130

For `statistics` it is desirable to override the defaults and only compute what the user supplies, so this works. But with `fmt_fn` we want to *combine* user-supplied formatting with the defaults. Does that make sense, and if so, do you see an obvious solution for that?

Reply:
Ohhhh, good point, it won't work in this situation because we're talking lists nested within a list: one level deeper than the similar argument inputs. 🤔